src/squeue.rs

```diff
@@ -297,3 +297,27 @@ impl Display for PushError {
 }
 
 impl Error for PushError {}
+
+impl Debug for Entry {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Entry")
+            .field("op_code", &self.0.opcode)
+            .field("flags", &self.0.flags)
+            .field("user_data", &self.0.user_data)
+            .finish()
+    }
+}
+
+impl Debug for SubmissionQueue<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        let mut d = f.debug_list();
+        let mut pos = self.head;
+        while pos != self.tail {
+            let entry =
+                Entry(unsafe { *self.queue.sqes.add((pos & self.queue.ring_mask) as usize) });
```

Comment:
we can get `&Entry` directly, thus avoiding a copy?

Reply:
Yes, done.